src/container/srv_container.c

```diff
@@ -2547,3 +2551,163 @@ ds_cont_oid_fetch_add(uuid_t poh_uuid, uuid_t co_uuid, uuid_t coh_uuid,
 	return rc;
 }
 
+/* Send the RPC from a DAOS server instance to the container service */
+int
+ds_cont_svc_set_prop(uuid_t pool_uuid, uuid_t cont_uuid,
+		     d_rank_list_t *ranks, daos_prop_t *prop)
+{
+	int				rc;
+	struct rsvc_client		client;
+	crt_endpoint_t			ep;
+	struct dss_module_info		*info = dss_get_module_info();
+	crt_rpc_t			*rpc;
+	struct cont_prop_set_in		*in;
+	struct cont_prop_set_out	*out;
+
+	D_DEBUG(DB_MGMT, DF_UUID": Setting container prop\n",
+		DP_UUID(cont_uuid));
+
+	rc = rsvc_client_init(&client, ranks);
+	if (rc != 0)
+		D_GOTO(out, rc);
+
+rechoose:
+	ep.ep_grp = NULL; /* primary group */
+	rc = rsvc_client_choose(&client, &ep);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": cannot find pool service: "DF_RC"\n",
+			DP_UUID(cont_uuid), DP_RC(rc));
+		D_GOTO(out_client, rc);
+	}
+
+	rc = cont_req_create(info->dmi_ctx, &ep, CONT_PROP_SET, &rpc);
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to create cont set prop rpc: %d\n",
+			DP_UUID(cont_uuid), rc);
+		D_GOTO(out_client, rc);
+	}
+
+	in = crt_req_get(rpc);
+	uuid_clear(in->cpsi_op.ci_hdl);
+	uuid_clear(in->cpsi_op.ci_pool_hdl);
+	uuid_copy(in->cpsi_op.ci_uuid, cont_uuid);
+	uuid_copy(in->cpsi_pool_uuid, pool_uuid);
+	in->cpsi_prop = prop;
+
+	rc = dss_rpc_send(rpc);
+	out = crt_reply_get(rpc);
+	D_ASSERT(out != NULL);
+
+	rc = rsvc_client_complete_rpc(&client, &ep, rc,
+				      out->cpso_op.co_rc,
+				      &out->cpso_op.co_hint);
+	if (rc == RSVC_CLIENT_RECHOOSE) {
+		crt_req_decref(rpc);
+		dss_sleep(1000 /* ms */);
+		D_GOTO(rechoose, rc);
+	}
+
+	rc = out->cpso_op.co_rc;
+	if (rc != 0) {
+		D_ERROR(DF_UUID": failed to set prop for container: %d\n",
+			DP_UUID(cont_uuid), rc);
+	}
+
+	crt_req_decref(rpc);
+out_client:
+	rsvc_client_fini(&client);
+out:
+	return rc;
+}
+
+void
+ds_cont_set_prop_handler(crt_rpc_t *rpc)
+{
+	int				 rc;
+	struct cont_svc			*svc;
+	struct cont_prop_set_in		*in = crt_reply_get(rpc);
+	struct cont_prop_set_out	*out = crt_reply_get(rpc);
+	struct rdb_tx			 tx;
+	uuid_t				 pool_uuid;
+	uuid_t				 cont_uuid;
+	struct ds_pool			*pool;
+	struct cont			*cont;
+	struct ds_cont_hdl		*hdl;
+	uuid_t				 hdl_uuid;
+
+	/* Client RPCs go through the regular flow with pool/cont handles */
+	if (daos_rpc_from_client(rpc)) {
+		ds_cont_op_handler(rpc);
+		return;
+	}
+
+	/*
+	 * Server RPCs don't have pool or container handles, so we need to fetch
+	 * the pool and container info.
+	 */
+	uuid_copy(pool_uuid, in->cpsi_pool_uuid);
+	uuid_copy(cont_uuid, in->cpsi_op.ci_uuid);
+
+	D_DEBUG(DF_DSMS, DF_CONT": processing cont set prop rpc %p\n",
+		DP_CONT(pool_uuid, cont_uuid), rpc);
+
+	rc = cont_svc_lookup_leader(pool_uuid, 0 /* id */,
+				    &svc, &out->cpso_op.co_hint);
+	if (rc != 0) {
+		D_ERROR("Failed to look up cont svc: %d\n", rc);
```

Comment:
In the container module, we try to print messages in the following format:

    DF_CONT": ..."

A minor issue; just FYI.

Reply:
Fixed.